### PR TITLE
[Java][Spring] fix @Operation content for array response

### PR DESCRIPTION
**Pull request by borsch**
Fix for https://github.com/OpenAPITools/openapi-generator/issues/14178

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@cachescrubber (2022/02) @welshm (2022/02) @MelleD (2022/02) @atextor (2022/02) @manedev79 (2022/02) @javisst (2022/02) @borsch (2022/02) @banlevente (2022/02) @Zomzog (2022/09)

### CODE REVIEW

**Review by msg2santoshsagar — APPROVED** [review]
Looks good..

**Comment by borsch** [comment]
@wing328 please review

File: samples/client/others/typescript-axios/with-separate-models-and-api-inheritance/common.ts [hunk 1]
```diff
@@ -85,6 +85,7 @@ export const setOAuthToObject = async function (object: any, name: string, scope
 }
 
 function setFlattenedQueryParams(urlSearchParams: URLSearchParams, parameter: any, key: string = ""): void {
+    if (parameter == null) return;
```
Comment (welshm) [inline samples/client/others/typescript-axios/with-separate-models-and-api-inheritance/common.ts:88]: This seems unrelated to what you changed?

File: modules/openapi-generator/src/main/resources/JavaSpring/api.mustache [hunk 2]
```diff
@@ -161,7 +162,7 @@ public interface {{classname}} {
             {{#responses}}
             @ApiResponse(responseCode = "{{{code}}}", description = "{{{message}}}"{{#baseType}}, content = {
                 {{#produces}}
-                @Content(mediaType = "{{{mediaType}}}", schema = @Schema(implementation = {{{baseType}}}.class)){{^-last}},{{/-last}}
+                @Content(mediaType = "{{{mediaType}}}", {{#isArray}}array = @ArraySchema( {{/isArray}}schema = @Schema(implementation = {{{baseType}}}.class){{#isArray}}){{/isArray}}){{^-last}},{{/-last}}
```
Comment (welshm) [inline modules/openapi-generator/src/main/resources/JavaSpring/api.mustache:165]: Does this require any version checks for Swagger? Or it should have always been there?

**Review by welshm — APPROVED** [review]
Nice!

File: modules/openapi-generator/src/main/resources/JavaSpring/api.mustache (hunk 2 above)
Comment (borsch) [reply modules/openapi-generator/src/main/resources/JavaSpring/api.mustache:165]: seems like it's in all versions of v3 package

https://github.com/swagger-api/swagger-core/blob/master/modules/swagger-annotations/src/main/java/io/swagger/v3/oas/annotations/media/Content.java#L70

File: samples/client/others/typescript-axios/with-separate-models-and-api-inheritance/common.ts (hunk 1 above)
Comment (borsch) [reply samples/client/others/typescript-axios/with-separate-models-and-api-inheritance/common.ts:88]: yeah, without this "samples up-to date" pipeline is failling

File: modules/openapi-generator/src/main/resources/JavaSpring/api.mustache (hunk 2 above)
Comment (cachescrubber) [reply modules/openapi-generator/src/main/resources/JavaSpring/api.mustache:165]: The while section is wrapped in  a `{{#swagger2AnnotationLibrary}}` Block. So it is only in effect with Swagger v2.x / OAS v3.

**Review by cachescrubber — APPROVED** [review]

**Comment by cachescrubber** [comment]
Related issue? #13332.

**Comment by borsch** [comment]
@wing328 could you please review and merge?

**Comment by wing328** [comment]
LGTM and I've not tested it locally
